kqueue.go

```diff
@@ -231,7 +235,16 @@ func (w *Watcher) addWatch(name string, flags uint32) (string, error) {
 
 	if !alreadyWatching {
 		w.mu.Lock()
+		parentName := filepath.Dir(name)
 		w.watches[name] = watchfd
+
+		watchesByDir, ok := w.watchesByDir[parentName]
+		if !ok {
+			watchesByDir = make(map[int]struct{})
```

Comment:
we can set the initial size of the map to 1 because we know we're going to insert one element.

Reply:
oh, good point!